lib/msal-node/src/cache/Storage.ts

```diff
@@ -194,8 +194,9 @@ export class Storage extends CacheManager {
      */
     getIdTokenCredential(idTokenKey: string): IdTokenEntity | null {
         const credType = CredentialEntity.getCredentialType(idTokenKey);
-        if (credType === CredentialType.ID_TOKEN) {
-            return this.getItem(idTokenKey) as IdTokenEntity;
+        const idToken = this.getItem(idTokenKey);
+        if (idToken && credType === CredentialType.ID_TOKEN) {
```

Comment:
Don't we have `isIdTokenEntity` we can use here instead?

Reply:
@pkanher617  and me plan to add the validation in a single PR for browser and node, hence did not do it here. But yes, plans are there to add it. Currently this change is needed as `null` or `undefined` items are being accessed when we do all key sweep and failing `silent` calls for `msal-node`

Reply:
validation was already added for browser, it might be good to add it here for node now.

Reply:
:-D Didn't know you already did; As discussed, will be adding this soon and some more tests.

Reply:
Made changes.